Instant-Messenger/src/util/user.cpp

```diff
@@ -32,17 +34,18 @@ void User::getActiveSockets(int* activeSocketsResult) {
     post_semaphore();
 }
 
-void User::registerSession(int socket) {
+int User::registerSession(int socket) {
     wait_semaphore();
     for (int i = 0; i < NUMBER_OF_SIMULTANEOUS_CONNECTIONS; i++) {
```

Comment:
Usar vector ao invés de vetor
Dar pushback de usuários sempre que vetor for de tamanho menor que 2
Se for tamanho 2, denega o novo usuário por ter numero maximo de conexões